ArcGISRuntimeSDKQt_CppSamples/Maps/GenerateOfflineMap_Overrides/GenerateOfflineMap_Overrides.cpp

```diff
@@ -0,0 +1,458 @@
+// [WriteFile Name=GenerateOfflineMap, Category=Maps]
+// [Legal]
+// Copyright 2017 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "GenerateOfflineMap_Overrides.h"
+
+#include "FeatureLayer.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "AuthenticationManager.h"
+#include "Portal.h"
+#include "PortalItem.h"
+#include "OfflineMapTask.h"
+#include "GeometryEngine.h"
+#include "Envelope.h"
+#include "Point.h"
+
+using namespace Esri::ArcGISRuntime;
+
+const QString GenerateOfflineMap_Overrides::s_webMapId = QStringLiteral("acc027394bc84c2fb04d1ed317aac674");
+
+GenerateOfflineMap_Overrides::GenerateOfflineMap_Overrides(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void GenerateOfflineMap_Overrides::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<GenerateOfflineMap_Overrides>("Esri.Samples", 1, 0, "GenerateOfflineMap_OverridesSample");
+  qmlRegisterUncreatableType<AuthenticationManager>("Esri.Samples", 1, 0, "AuthenticationManager", "AuthenticationManager is uncreateable");
+}
+
+void GenerateOfflineMap_Overrides::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
+
+  // Create a Portal Item for use by the Map and OfflineMapTask
+  bool loginRequired = true;
+  Portal* portal = new Portal(loginRequired, this);
+  m_portalItem = new PortalItem(portal, webMapId(), this);
+
+  // Create a map from the Portal Item
+  m_map = new Map(m_portalItem, this);
+
+  // Update property once map is done loading
+  connect(m_map, &Map::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+      return;
+
+    m_mapLoaded = true;
+    emit mapLoadedChanged();
+  });
+
+  // Set map to map view
+  m_mapView->setMap(m_map);
+
+  // Create the OfflineMapTask with the PortalItem
+  m_offlineMapTask = new OfflineMapTask(m_portalItem, this);
+
+  // connect to the error signal
+  connect(m_offlineMapTask, &OfflineMapTask::errorOccurred, this, [](Error e)
+  {
+    if (e.isEmpty())
+      return;
+
+    qDebug() << e.message() << e.additionalMessage();
+  });
```

Comment:
Somewhat conflicted about this one. I think it is worthwhile logging these things for people that try to run the sample independently. 

We would also need to remove it from the initial "Generate Offline Map" sample as well. Thoughts?

Reply:
Yeah, I was thinking the same thing after I commented on these.
@ldanzinger what do you think?  Do we have these print statements for errors connected up in other samples, or are there any guidelines for doing so?

Although we don't necessarily print any of these errors to the UI, it could be handy for a developer who is copying code out of the sample to begin their own app.

Reply:
I always think that the errors (and handling them) is a central part of our API and so we should show them front and center. It also makes trouble shooting easier

Reply:
Agree.  I guess if we all feel this way then the samples should all handle errors in the UI properly?  Right now just printing them out to the console doesn't seem "front and center" to me.  Seems a little half baked.  But I would like to know if there are any guidelines from the sample feature team standpoint as to what we should be doing.

Reply:
We don't really have an official standpoint. Probably would be a good thing to talk about. I think this is ok to leave in.

Reply:
Cool, let's leave them in